content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +690,772 @@ as expected.
 
 Since no matter the size of the submatrix we are summing, we only need to access four values of the 2D prefix sum array, this runs in $\mathcal{O}(1)$ per query after an $\mathcal{O}(NM)$ preprocessing.
 
+## Creation From Endpoints
+
+<FocusProblem problem={problems.sample3[0]} />
+
+### 1-Dimensional
+
+A common application of prefix sums is to determine, given a set of intervals, the number of intervals that cover any particular point.
+
+In the focus problem, we are given $n$ intervals $[a,b]$, and we want to find the maximum number of intervals that cover any particular point.
+
+A naïve approach to this problem is to create an array $\texttt{ctr}$ where $\texttt{ctr}$ is the number of intervals that cover each point $i$. We can do this by looping through each interval $[a,b]$ and increasing $\texttt{ctr}[i]$ by $1$ for each index in $a \leq i \leq b$.
+
+This results in a $\mathcal{O}(nV)$ time complexity (where $0 \leq a < b \leq V$), which easily times out (think what happens when the interval $[0, V]$ is queried $n$ times).
+
+We can do better. It's easy to see that an increment of $x$ (before computation) in $\texttt{arr}[i]$ causes all subsequent $\texttt{prefix}[i...N]$ (after computation) to increase by $x$. This concept can be conceptualized through increment and decrement points. An increment point increases (and decrement decreases) all subsequent cells. (Note that our decrement point is located at $B+1$ because the interval is inclusive).
+
+Example: Add two to each point in the interval $[2, 4]$
+
+Our matrix after adding 2 at our increment (start) point (before computation)
+
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td className='bg-green-200 dark:bg-green-800'>2</td>
+			<td>0</td>
+			<td>0</td>
+			<td>0</td>
+		</tr>
+	</table>
+</center>
+
+Our matrix after adding 2 at our increment (start) point (and computing).
+
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td className='bg-green-200 dark:bg-green-800'>2</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+		</tr>
+	</table>
+</center>
+
+Our matrix after subtracting 2 at our decrement point (and computing).
+
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+			<td className='bg-red-200 dark:bg-red-800'>0</td>
+		</tr>
+	</table>
+</center>
+
+In this above problem, we are only incrementing by 1, so our $x$ is consistent. As a result, when we encounter a starting point, we can increment by 1, and for an endpoint, decrement by 1. For our problem, we don't need to include a prefix array because we aren't querying a range, so a single variable suffices.
+
+Observe that this works for multiple intervals.
```

Comment:
Reorder these last two sections (put implementation details after the explanation)